### DISCUSSION TITLE
Setting up a VS 2019 PS shell?

### DISCUSSION THREAD

**Discussion by rfancher** [Eugeny/tabby#4118] [Q&A]
Hi, I'm having a hard time setting up a Visual Studio 2019 Powershell configuration.

My default shortcut created by the ID has the following at the target:
C:\Windows\SysWOW64\WindowsPowerShell\v1.0\powershell.exe -noe -c "&{Import-Module """C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"""; Enter-VsDevShell 3fa88563}"

When I attempt to setup a shell profile with this:

Command: `powershell.exe`
Arguments: `-noe -c "&{Import-Module """C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"""; Enter-VsDevShell 3fa88563}"`

I eventually get this error:
```
Enter-VsDevShell : Item has already been added. Key in dictionary: 'COMSPEC'  Key being added: 'ComSpec'
At line:1 char:135
+ ... ools\Microsoft.VisualStudio.DevShell.dll"; Enter-VsDevShell 3fa88563}
+                                                ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Enter-VsDevShell], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.VisualStudio.DevShell.Commands.EnterVsDev 
   ShellCommand
```

I can get it working in a regular cmd.exe window - I'm not getting why it wont work here, would appreciate any help, thanks!

**Comment by Eugeny** [comment]
The automatic COMSPEC setting is now optional and disabled by default in v145 - please see if that helps

**Reply by rfancher** [reply]
That setting helped move past that error, but still seem to have problems setting up the profile for this shell.  Unfortunately it closes too fast for me to see the error messages.  Is there a way to see it in the log?
When I try to debug the arguments in a built in shell it all seems to work, but something about the syntax/format in the profile causes an error.

**Reply by Eugeny** [reply]
Yep - it's not in the UI, but you can set options.pauseAfterExit: true on your profile in the config file.
